### PR TITLE
Tf 1489: create and implement of ParseCalenderMethod

### PR DESCRIPTION
**Pull request by sakriwassim**

### CODE REVIEW

File: lib/jmap/calendar/calender_event.dart [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        method,
+        sequence,
+        priority,
+        freeBusyStatus,
+        privacy,
+        organizer,
+        participants,
+        extensionFields,
+        recurrenceRules
+      ];
```
Comment (nqhhdev) [inline lib/jmap/calendar/calender_event.dart:76]:
```suggestion
  @override
  List<Object?> get props => [
    uid,
    title,
    description,
    start,
    duration,
    end,
    timeZone,
    location,
    method,
    sequence,
    priority,
    freeBusyStatus,
    privacy,
    organizer,
    participants,
    extensionFields,
    recurrenceRules
  ];
```

File: lib/jmap/calendar/participants.dart [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+      this.expectReply});
+
+  factory Participants.fromJson(Map<String, dynamic> json) =>
+      _$ParticipantsFromJson(json);
+
+  Map<String, dynamic> toJson() => _$ParticipantsToJson(this);
+
+  @override
+  List<Object?> get props =>
+      [name, mailto, kind, role, participationStatus, expectReply];
```
Comment (nqhhdev) [inline lib/jmap/calendar/participants.dart:30]:
```suggestion
  List<Object?> get props => [
    name, 
    mailto, 
    kind, 
    role, 
    participationStatus, 
    expectReply
  ];
```

File: lib/jmap/calendar/participants.dart [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  final String? participationStatus;
+  final bool? expectReply;
+
+  Participants(
+      {this.name,
+      this.mailto,
+      this.kind,
+      this.role,
+      this.participationStatus,
+      this.expectReply});
```
Comment (nqhhdev) [inline lib/jmap/calendar/participants.dart:21]:
```suggestion
  Participants({
    this.name,
    this.mailto,
    this.kind,
    this.role,
    this.participationStatus,
    this.expectReply});
```

File: lib/jmap/calendar/recurrence_rules.dart [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  final List<String>? byMonth;
+  final List<int>? bySetPosition;
+  final String? until;
+
+  RecurrenceRules(
+      {this.frequency,
+      this.byDay,
+      this.byMonth,
+      this.bySetPosition,
+      this.until});
```
Comment (nqhhdev) [inline lib/jmap/calendar/recurrence_rules.dart:19]:
```suggestion
  RecurrenceRules({
    this.frequency,
    this.byDay,
    this.byMonth,
    this.bySetPosition,
    this.until});
```

File: lib/jmap/core/capability/capability_identifier.dart [hunk 5]
```diff
@@ -10,7 +10,7 @@ class CapabilityIdentifier with EquatableMixin {
   static final jmapQuota = CapabilityIdentifier(Uri.parse('urn:ietf:params:jmap:quota'));
   static final jmapTeamMailboxes = CapabilityIdentifier(Uri.parse('urn:apache:james:params:jmap:mail:shares'));
   static final jamesSortOrder = CapabilityIdentifier(Uri.parse('urn:apache:james:params:jmap:mail:identity:sortorder'));
-
+  static final calendarEvent = CapabilityIdentifier(Uri.parse('com:linagora:params:calendar:event'));
```
Comment (nqhhdev) [inline lib/jmap/core/capability/capability_identifier.dart:13]: static final jmapCalendarEvent = CapabilityIdentifier(Uri.parse('...'));

File: lib/jmap/core/method/request/get_method.dart [hunk 6]
```diff
@@ -6,15 +6,36 @@ import 'package:jmap_dart_client/jmap/core/request/result_reference.dart';
 import 'package:json_annotation/json_annotation.dart';
 
 abstract class GetMethod extends MethodRequiringAccountId
-    with OptionalIds, OptionalProperties, OptionalReferenceIds, OptionalReferenceProperties {
+    with
+        AdditionalBlobIds,
+        OptionalIds,
+        OptionalProperties,
+        OptionalReferenceIds,
+        OptionalReferenceProperties {
```
Comment (nqhhdev) [inline lib/jmap/core/method/request/get_method.dart:14]:
```suggestion
    with AdditionalBlobIds, OptionalIds, OptionalProperties,
      OptionalReferenceIds, OptionalReferenceProperties {
```

File: lib/jmap/core/method/request/get_method.dart [hunk 7]
```diff
@@ -6,15 +6,36 @@ import 'package:jmap_dart_client/jmap/core/request/result_reference.dart';
...
 }
 
 abstract class GetMethodNoNeedAccountId extends Method
-    with OptionalIds, OptionalProperties, OptionalReferenceIds, OptionalReferenceProperties {
+    with
+        AdditionalBlobIds,
+        OptionalIds,
+        OptionalProperties,
+        OptionalReferenceIds,
+        OptionalReferenceProperties {
```
Comment (nqhhdev) [inline lib/jmap/core/method/request/get_method.dart:24]: same as above

File: lib/jmap/core/method/request/get_method.dart [hunk 8]
```diff
@@ -6,15 +6,36 @@ import 'package:jmap_dart_client/jmap/core/request/result_reference.dart';
...
   GetMethodNoNeedAccountId() : super();
 }
 
+mixin AdditionalBlobIds {
+  List<String>? blobIds;
+  void addString(String newString) {
```
Comment (nqhhdev) [inline lib/jmap/core/method/request/get_method.dart:30]: Add space

File: test/jmap/calendar/blob_found_not_parsable_test.dart [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+            "c1"
+          ]
+        ]
+      };
+
+      test('blob found not parsable', () async {
+        final baseOption = BaseOptions(method: 'POST');
+        final dio = Dio(baseOption)
+          ..options.baseUrl =
+              "https://gateway.upn.integration-open-paas.org"; //'http://domain.com/jmap';
```
Comment (nqhhdev) [inline test/jmap/calendar/blob_found_not_parsable_test.dart:49]:
```suggestion
          ..options.baseUrl = "https://gateway.upn.integration-open-paas.org";
```
If you do not use code pls delete it before pushing it

File: test/jmap/calendar/blob_found_not_parsable_test.dart [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+
+        final httpClient = HttpClient(dio);
+        final processingInvocation = ProcessingInvocation();
+        final jmapRequestBuilder =
+            JmapRequestBuilder(httpClient, processingInvocation);
+        final accountId = AccountId(Id(
+            '0eacc7a5c74b27ab36a823bc5c34da36e16c093705f241d6ed5f48ee73a4ecfb'));
+        // final result = await (jmapRequestBuilder).build().execute();
+
+        // expect(result, response1);
```
Comment (nqhhdev) [inline test/jmap/calendar/blob_found_not_parsable_test.dart:66]: Do you use it here?

**Comment by nqhhdev** [comment]
Check why it failed again, fix it before you finish PR

Screenshot 2023-04-17 at 00 12 39

**Comment by hoangdat** [comment]
You can run your test cases directly in your IDE or in terminal.

**Comment by hoangdat** [comment]
Remove all unrelated `*.g` files.

**Comment by dab246** [comment]
Continue at #68

**Pull request closed without merging**